text/0186-per-tracer-configuration.md

```diff
@@ -0,0 +1,124 @@
+  
+- Provides a way to disable Tracing in a library or for any Named Tracer.
+
+This is already mentioned in the specification:
+
+> If configuration must be stored per-tracer (such as disabling a certain
+> tracer), the tracer could, for example, do a look-up with its
+> name+version+schema_url in a map in the TracerProvider, or the TracerProvider
+> could maintain a registry of all returned Tracers and actively update their
+> configuration if it changes.
+
+But with no requirement or API related to actually providing this feature.
+
+## Explanation
+
+Tracer instances are created through a Tracer Provider which stores the
+configuration used for each Tracer (i.e., SpanProcessors, IdGenerator,
+Span Limits and Sampler). Creating or retrieving a Tracer from the Tracer
+Provider requires a name (and optional version) that identify the
+instrumentation library.
+
+In addition to the default configuration, the Tracer Provider can have separate
+configuration settings for individual Tracers based on the name of the Named
+Tracer. Adding a configuration does not require the Named Tracer to exist, the
+name and/or version, is the only requirement.
+
+For example, a configuration with Sampler of `always_off` can be added under the
+name `io.opentelemetry.contrib.mongodb`. Then, when a Tracer is created through
+the Tracer Provider with name `io.opentelemetry.contrib.mongodb` it will use the
+`always_off` Sampler, but the rest of its configuration is the same as the
+default Tracer Provider configuration.
+
+## Internal details
+
+Where possible the configuration should remain shared between Tracer instances
+as a reference to avoid allocations.
+
+Configuration should be able to be done both through static configuration and at
+the time a Tracer Provider is created.
+
+## Trade-offs and mitigation
+
+- Additional memory usage and allocation.
+
+Before this change each Tracer could hold a reference to a shared configuration
+provided by the Tracer Provider. But this will only be the case when a different
+configuration is used for a Tracer, all others can remain references.
+
+## Prior art and alternatives
+
+- Multiple Tracer Providers
+
+The specification as of this writing hints at the use of multiple Tracer
+Providers in order to allow the creation of Tracers with different
+configurations:
+
+> some applications may want to or have to use multiple TracerProvider
+> instances, e.g. to have different configuration (like SpanProcessors) for each
+> (and consequently for the Tracers obtained from them)
+
+While this proposal does not require removing this capability it does replace
+the necessity of using multiple Tracer Providers in order to have different
+configurations. Most existing implementations do not offer the
+capability of creating multiple Tracer Providers and may essentially prohibit
+it. Allowing configuration to be done through the Tracer Provider based on the
```

Comment:
Is this true?  I don't recall anything in the spec that would lead to implementations that are incapable of supporting multiple tracer providers.  While multiple providers do present the issue noted below WRT sharing exporters, I think that any problem caused by sharing exporters can be solved without new specification (i.e., treated as implementation details).

Reply:
There isn't anything in the spec that would necessarily lead to it. Some, like Java, support multiple providers only through having multiple SDKs.

I don't think leaving how exporters are shared up to each implementation would be a nice user experience, and definitely not lead to consistent support.